### ISSUE TITLE
Feature Request - additional power objects

### ISSUE THREAD

**Issue by sOckhamSter** [ulic75/power-flow-card#67]
Allow two (for now) additional objects hanging off the Home. In my example image, these are an electric car and an electric water heater, but in my mind they would be customisable entities that you could label as whatever you like (for example, the water heater could show water temperature), and choose whatever icons you like.
![Power Flow Card Ideas](https://user-images.githubusercontent.com/4546972/169845770-9c796d64-bd82-46f1-8d03-6225046fdc23.jpg)

**Comment by GamerClassN7** [comment]
Is there any progress on this ?

**Comment by burrellka** [comment]
I can have a go at it if someone can point me to how to get the repo setup and a valid test environment. I took a look at the raw jss and I could be wrong but it didnt seem at all that difficult. I love this template/card and give the author a lot of credit. It's the best and cleanest out there imo. My own perosnal enhancements I'd like to see is the ability to add current water consumption (I use FLOW water meter sensor with HA) and to have current/day gas consumption (currently building gas reading sensing using AI on the edge

**Comment by sOckhamSter** [comment]
I'm on a Mac, so I used Homebrew to install npm / pnpm first, then I installed VSCode. Once that's installed, on first run it has an option to clone a Github repo so just paste in the URL for it and let it clone. Then you should be able to open a console in VScode and type 'npm run build' which generates the .js that you need to use in Home Assistant's config/www folder. This might all be very obvious to seasoned developers, but I was new to this particular development environment!

**Comment by BadgerLoaf** [comment]
@ulic75 - What are your thoughts on this request?
I prefer your card over anything else out there, it looks so much more polished. But I will admit that the ability to add a couple of additional objects is something that I do miss compared to the other options available. 

Being able to display the usage of specific high usage objects would be very useful.

Out of interest, what is the opinion of people on whether the home object should display the total home usage, or the home usage minus the 1 or 2 additional objects? I can see a case for either configuration.

**Comment by spali** [comment]
@BadgerLoaf 
Couldn't write my mind better.

> PS: Out of interest, what is the opinion of people here on whether the home object should display the total home usage, or the home usage minus the 1 or 2 additional objects? I can see a case for either configuration.
> 

In my mind, it depends on the solution, how the objects are connected.
If completely separate, then the home without the additionals.
But personally I would prefer the example of OP, with the objects connected to the home (kind as sub objects of home) and then I would expect home to show the total.

**Comment by ulic75** [comment]
@BadgerLoaf To be completely honest, my life is just super busy right now.  I've contemplated in my head writing up some kind of official goal for the project stating "Mirror the design of the Official Energy Distribution card, without many extra".  I know that would disappoint many people, but would at least manage expectations.  Frankly if someone wanted to fork my work into a new project and add this stuff I'd be totally fine with it.  I made the card to fill my need and then decided to share it.  Sadly that's probably the death of a lot of projects, but it's the truth.  Maybe someday I'll get back to it, but for now I'm just lacking time and motivation.

**Comment by BadgerLoaf** [comment]
@ulic75
Fully understand and empathise. I understand the card was created for your needs, and you were kind enough to share with the community, so thank you. I also know the amount of time these things can take... which when balanced with work and home life can rarely balance.

Maybe someone (with the skills I don't have !) will pick this up and fork it, or you will decide to look at this again in time, but for now thanks for sharing... it is a great card.

**Comment by flyrmyr** [comment]
Based on @ulic75 response, it seems like this repo won't see much movement soon. We definitely appreciate all the work you've put in!

Tossing this out there, I'm working on a more feature rich fork of this repo with a bit of an angled take... I'm an 8 year software dev, brand spankin new to open source as well as Hassio, living in a van (hence my example), trying to solve for my own use case and make this useful for others.

https://github.com/flyrmyr/power-flow-card
** for now
Feel free to grab the system_flow_card.js from the ./dist folder and see if it helps y'all.

Aiming for: 
System/unit/resource agnostic flow chart with this design (ie. including but not solely power/energy). 
It should hit all linked feature requests to this issue: 
additional elements
custom icons
circle charted
custom "extra value(s?)"
custom colors
Combining the in/out to one line, hiding the arrow if net zero?
Ideally, visually, maximum 9 main producers/consumers, and 3 "breakout" elements off of the base system element. This is what I have so far, with the current design for comparison:
<img width="502" alt="Screen Shot 2022-11-04 at 2 56 06 AM" src="https://user-images.githubusercontent.com/14035884/199934911-a0d847ee-823d-4da1-966b-6c7302f98429.png">
The yaml currently supports:
<img width="549" alt="Screen Shot 2022-11-04 at 2 56 18 AM" src="https://user-images.githubusercontent.com/14035884/199936365-9fec2c54-b1c8-4b3f-bd56-bf1ba9cef1d0.png">
Maintaining or adding the option for in & out from each element.

Basically reworking my current Victron screen for HA:
<img width="485" alt="Screen Shot 2022-11-04 at 3 32 52 AM" src="https://user-images.githubusercontent.com/14035884/199941285-73369b56-e856-446b-86d3-6b97fcd59c1d.png">


Big thanks to @sOckhamSter on the dev env hints. Still WIP but I have no idea how to get HACS to "pass" my forked repo as a Custom Repository without tossing the "Repository structure for XXX is not compliant" error. A big help would be anyone with domain knowledge of HACS who could detail why this fails (I've been through the HACS docs to create a viable repo).

And really any other feature requests that fit into the agnostic design are welcome as well. Again, new to the open source community, but reach out however is standard and I'll do my best.

**Comment by BadgerLoaf** [comment]
@flyrmyr - Thanks for sharing your effort on this, and your uptake of many of the requests on here. 

Unfortunately I can't help on the HACS queries, but hopefully someone can.

Once you have this available to install, I'll happily install and try it out.

Thanks!

**Comment by flyrmyr** [comment]
For anyone following this, I've reworked this fully to include:
- any number of system "elements" including loads/vehicle charging etc. with custom positioning
- defining icons, colors and flow speed (open to custom flow algorithm if needed)
- custom "extra" data shown per element
- faded idle elements

Systems could be power, energy, water/fluid, internet device traffic - you name it / you define it!

My personal use case is shown here, above the current design (my van being the "home" element, displaying any leftover units in the system - in this case only my inverter's un accounted for internal draw):
<img width="452" alt="Screen Shot 2022-11-11 at 3 30 47 AM" src="https://user-images.githubusercontent.com/14035884/201321775-8236b349-17fd-4310-a0b4-33b1d29dad0e.png">
with the yaml file:
<img width="400" alt="Screen Shot 2022-11-11 at 3 38 18 AM" src="https://user-images.githubusercontent.com/14035884/201323111-0a0fab36-6dab-46fe-8806-d80108701a62.png">

This should still provide equal insight as the current power/energy panels, plus a tonnn of extra. Please, if anyone has any additional requests, send them over, happy to accommodate while I'm in active development. Still definitely planning on getting this into HACS if anyone has experience getting it through their guidelines barrier.
 
** Any testers, just drop the file from `./dist` into `./config/www` and this is useable as-is.

**Comment by sOckhamSter** [comment]
That looks REALLY awesome!

**Comment by BadgerLoaf** [comment]
@flyrmyr - Looking great Eric! Will try it out this weekend. Personally I love the faded idle items.... but out of interest, is that an option that can be toggled? Just thinking for people who might prefer not having them faded when idle.

Thanks again and I'll let you know how I get on

**Comment by flyrmyr** [comment]
Thanks!
@BadgerLoaf good call, added a flag for `fadeIdyeElements` and it's off by default.

**Comment by BadgerLoaf** [comment]
@flyrmyr i haven’t had a chance to look in-depth, and will do later, but not sure I follow your instructions on how to install this.

Sorry, not manually installed before.

**Comment by burrellka** [comment]
This is amazing!! flyrmyr thank you so much for building on ulic75's great work. I'm looking forward to testing that out this evening.

**Comment by BadgerLoaf** [comment]
Hi @flyrmyr - Apologies if I'm missing something, but where can I find the ```./dist``` folder?

**Comment by BadgerLoaf** [comment]
> Thanks! @BadgerLoaf good call, added a flag for `fadeIdyeElements` and it's off by default.

Cool, was just a thought!

**Comment by burrellka** [comment]
I did not. I setup my development environment, and compiled the repo to JS. That all worked, but then i added the JS to me www/custom folder, restarted HA and it didnt show up to be selected. thats where i got confused. I got it built, how do i get the custom component to be selectable.  I then figured i could maybe replace the official power flow js in the custom folder, but I didnt want to do that since i have that officially configured and working on my dashboard.

**Comment by BadgerLoaf** [comment]
@burrellka - I assume you added the resource reference?

<img width="839" alt="Screenshot 2022-11-14 at 21 20 21" src="https://user-images.githubusercontent.com/108699847/201768039-5f4f6e57-d3c8-4b2b-a2db-74036896aa7e.png">

**Comment by BadgerLoaf** [comment]
@flyrmyr - Any chance you could help and make the file(s) available to test? Thanks!

**Comment by ulic75** [comment]
While I appreciate the innovation and the beautiful card you've designed.  Would mind taking the conversation else so I'm not sent all the updates that don't directly have anything to do with this card.

**Comment by BadgerLoaf** [comment]
Ofcourse @ulic75, apologies! Thanks again for all the work on your card.

@flyrmyr and @burrellka .... I have created a query about this here https://github.com/flyrmyr/system-flow-card/issues/1
